src/components/Api.js

```diff
@@ -0,0 +1,52 @@
+export class Api {
+    constructor(options) {
+        this._baseUrl = options.baseUrl;
+        this._headers = options.headers;
+    }
+
+    async _request(url, method, requestBody) {
+        return fetch(this._baseUrl + url, {
+            method: method,
+            headers: this._headers,
+            body: JSON.stringify(requestBody)
+          })
+            .then((res) => {
+                if (res.ok) {
+                    return res.json()
+                }
+            })
+            .catch((err) => console.log("Error: " + err));
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.